### PR TITLE
GSKIT some fixes

### PR DESCRIPTION
**Pull request by AKuHAK**
Fixed Makefile to be a little bit more up-to-date with upstream ps2sdk.
now gskit is compilable under gcc.=v5.3.0
Fixed some warnings and errors

### CODE REVIEW

File: ee/Rules.make [hunk 1]
```diff
@@ -32,37 +34,48 @@ EE_ASFLAGS := -G0 $(EE_ASFLAGS)
...
+EE_CXX_COMPILE = $(EE_CXX) $(EE_CXXFLAGS) $(EE_INCS)
 
+# Extra macro for disabling the automatic inclusion of the built-in CRT object(s)
+ifeq ($(EE_CC_VERSION),3.2.2)
+	EE_NO_CRT = -mno-crt0
+else ifeq ($(EE_CC_VERSION),3.2.3)
+	EE_NO_CRT = -mno-crt0
+else
+	EE_NO_CRT = -nostartfiles
+endif
```
Comment (rickgaiser) [inline ee/Rules.make]: Why is this code re-added? See [this commit](https://github.com/ps2dev/ps2toolchain/commit/48ba5757adcf04371dbd3d34174b167f63b6954f). You probably need to add the `#define LIB_SPEC` to gcc v5.3.0.

File: ee/Rules.make [hunk 2]
```diff
@@ -32,37 +34,48 @@ EE_ASFLAGS := -G0 $(EE_ASFLAGS)
...
-	$(EE_CXX) $(EE_LDFLAGS) -o $(EE_BIN) $(EE_OBJS) $(EE_LIBS)
+$(EE_BIN) : $(EE_OBJS) $(PS2SDK)/ee/startup/crt0.o
+	$(EE_CXX) $(EE_NO_CRT) -T$(PS2SDK)/ee/startup/linkfile $(EE_CFLAGS) \
+		-o $(EE_BIN) $(PS2SDK)/ee/startup/crt0.o $(EE_OBJS) $(EE_LDFLAGS) $(EE_LIBS)
 else
-$(EE_BIN): $(EE_OBJS)
-	$(EE_CC) $(EE_LDFLAGS) -o $(EE_BIN) $(EE_OBJS) $(EE_LIBS)
+$(EE_BIN) : $(EE_OBJS) $(PS2SDK)/ee/startup/crt0.o
+	$(EE_CC) $(EE_NO_CRT) -T$(PS2SDK)/ee/startup/linkfile $(EE_CFLAGS) \
+		-o $(EE_BIN) $(PS2SDK)/ee/startup/crt0.o $(EE_OBJS) $(EE_LDFLAGS) $(EE_LIBS)
```
Comment (rickgaiser) [inline ee/Rules.make]: same here.

File: examples/Makefile.global [hunk 3]
```diff
@@ -38,7 +38,7 @@ endif
 
 EE_LIBS += -lgskit_toolkit
 EE_LIBS += -lgskit -ldmakit
-
+EE_LIBS += -lc -lkernel
```
Comment (rickgaiser) [inline examples/Makefile.global]: same here.

File: ee/Rules.make [hunk 4]
```diff
@@ -32,37 +34,48 @@ EE_ASFLAGS := -G0 $(EE_ASFLAGS)
...
 $(EE_OBJS_DIR):
-	mkdir $(EE_OBJS_DIR)
+	mkdir -p $(EE_OBJS_DIR)
 
 ifeq ($(use_cpp), true)
-$(EE_BIN): $(EE_OBJS)
-	$(EE_CXX) $(EE_LDFLAGS) -o $(EE_BIN) $(EE_OBJS) $(EE_LIBS)
+$(EE_BIN) : $(EE_OBJS) $(PS2SDK)/ee/startup/crt0.o
+	$(EE_CXX) $(EE_NO_CRT) -T$(PS2SDK)/ee/startup/linkfile $(EE_CFLAGS) \
+		-o $(EE_BIN) $(PS2SDK)/ee/startup/crt0.o $(EE_OBJS) $(EE_LDFLAGS) $(EE_LIBS)
```
Comment (rickgaiser) [inline ee/Rules.make]: same here.

Comment (AKuHAK) [reply ee/Rules.make]: Ok I will try. did you have new gcc patches stored somewhere or I should try it from scratch?

Comment (rickgaiser) [reply ee/Rules.make]: No, I don't have the separate gcc v3.2.3 patches anymore. It should be a the combination of the following 2 commits in ps2toolchain:
https://github.com/ps2dev/ps2toolchain/commit/7c494f217d379639aaca23d3c588e48986177a51
https://github.com/ps2dev/ps2toolchain/commit/48ba5757adcf04371dbd3d34174b167f63b6954f

It's a single line to gcc and 3 lines to binutils.

Plus this fixup in the scripts:
`ln -sf "$PS2SDK/ee/startup/crt0.o"  "$PS2DEV/ee/lib/gcc-lib/ee/3.2.3/crt0.o"`
Where 3.2.3 probably needs to be changed to 5.3.0.

**Pull request closed without merging**